dynamodb-enhanced/deployment/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/EnhancedProcessor.java

```diff
@@ -0,0 +1,200 @@
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.BytecodeTransformerBuildItem;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.gizmo.Gizmo;
+import io.quarkus.runtime.LaunchMode;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
+
+public class EnhancedProcessor {
+    private static final String FEATURE = "amazon-dynamodb-enhanced";
+    public static final String CLASS_NAME_BEAN_TABLE_SCHEMA = "software.amazon.awssdk.enhanced.dynamodb.mapper.BeanTableSchema";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    AdditionalBeanBuildItem producer() {
+        return AdditionalBeanBuildItem.unremovableOf(DynamodbEnhancedClientProducer.class);
+    }
+
+    @BuildStep
+    public void registerClassesForReflectiveAccess(
+            CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        for (AnnotationInstance i : combinedIndexBuildItem
+                .getIndex()
+                .getAnnotations(DotName.createSimple(DynamoDbBean.class.getName()))) {
+            ClassInfo classInfo = i.target().asClass();
+            reflectiveClass.produce(
+                    new ReflectiveClassBuildItem(true, false, classInfo.name().toString()));
+        }
+
+        // When we add DefaultAttributeConverterProvider and BeanTableSchemaAttributeTags here, we do not need the DynamodbEnhancedFeature in runtime module.
+        //reflectiveClass.produce(
+        //        new ReflectiveClassBuildItem(true, false,
+        //                "software.amazon.awssdk.enhanced.dynamodb.DefaultAttributeConverterProvider"));
+        //reflectiveClass.produce(
+        //        new ReflectiveClassBuildItem(true, false,
+        //                "software.amazon.awssdk.enhanced.dynamodb.internal.mapper.BeanTableSchemaAttributeTags"));
+
+    }
+
+    @BuildStep
+    void maybeApplyClassTransformation(
+            BuildTimeConfig buildTimeConfig,
+            BuildProducer<BytecodeTransformerBuildItem> transformers,
+            LaunchModeBuildItem launchModeBuildItem) {
+        if (shouldApplyClassTransformation(buildTimeConfig, launchModeBuildItem)) {
+            applyClassTransformation(transformers);
+        }
+    }
+
+    private void applyClassTransformation(BuildProducer<BytecodeTransformerBuildItem> transformers) {
+        transformers.produce(
+                new BytecodeTransformerBuildItem(
+                        CLASS_NAME_BEAN_TABLE_SCHEMA, new MethodCallRedirectionVisitor()));
```

Comment:
Why are you using a constant instead of a reference to the class? It's not accessible?

Reply:
You are right, this shouldn't be necessary...